sigmastate/src/main/scala/sigmastate/UncheckedTree.scala

```diff
@@ -102,7 +102,7 @@ case class COrUncheckedNode(override val challenge: Challenge,
 
 case class CThresholdUncheckedNode(override val challenge: Challenge,
                                    override val children: Seq[UncheckedSigmaTree],
-                                   k: Integer,
+                                   k: Int,
```

Comment:
Why this change?

Reply:
As discussed above, we should avoid using boxed numerics.

Reply:
ah, right, Int will be unboxed, good catch.